src/libFLAC/cpu.c

```diff
@@ -251,7 +251,7 @@ ppc_cpu_info (FLAC__CPUInfo *info)
 	} else if (getauxval(AT_HWCAP2) & PPC_FEATURE2_ARCH_2_07) {
 		info->ppc.arch_2_07 = true;
 	}
-#elif defined(__FreeBSD__) && defined(HAVE_GETAUXVAL)
+#elif defined(__FreeBSD__)
```

Comment:
Shouldn't this be the following?

```suggestion
#elif defined(__FreeBSD__) && defined(HAVE_SYS_AUXV_H)
```

Reply:
Corrected.
However, elf_aux_info() is available since FreeBSD 12.0. Branches before 12 are already unsupported and frankly, so much happened in the powerpc* land in FreeBSD 13.0 that no one should use anything before 13.0.

Reply:
The thing is, that HAVE_SYS_AUXV_H check was added for uclibc here: https://github.com/xiph/flac/pull/292 I made the mistake to suggest to check for HAVE_GETAUXVAL, I see now that HAVE_SYS_AUXV_H must be checked.

I really don't know anything about uclibc nor FreeBSD, it seems uclibc is only compatible with linux, but perhaps other libc's have something similar, omitting sys/auxv.h on FreeBSD as well? In that case, this check would make sense even on FreeBSD 12 and later?

Reply:
I don't think there is any other libc available on FreeBSD for powerpc*. There actually are GNU/kFreeBSD distros (FreeBSD kernel with GNU userland), but since they use GNU, they probably use glibc (and all of them are only for amd64 / i386 anyway). I don't think anyone has ever attempted to run uclibc on FreeBSD and the probability of that happening on powerpc* is even lower.

I think you can just assume that FreeBSD 12.0 and newer have sys/auxv.h.

Reply:
Do you think we can leave it in though, just in case?

Reply:
Sure you can, it doesn't hurt.